ibis/mapd/operations.py

```diff
@@ -0,0 +1,798 @@
+from six import StringIO
```

Comment:
It looks like a lot of the functions and objecs in here are duplicated from either the impala or bigquery backends. Can you see if you can reuse some of their functions so we have only what's needed for MapD?

Reply:
Ok. I will take a look :) thanks!